### ISSUE TITLE
Telnet to SSL port = hang until disconnect

### ISSUE THREAD

**Issue by emurphy42** [brazilofmux/tinymux#694]
MUX 2.10.1.12 #9 [2012-JUL-07]

There's one regular port and one SSL port. Hitting the latter with a regular telnet connection causes the game session to hang. It generally resumes once the connection is dropped, but there have been a couple cases where it stays hung for minutes/hours, presumably because it didn't register the connection dropping. I could just turn SSL off (few actually use it) but I'd rather fix it.

- Is this a known issue?
- Is it fixed in a later version?
- Is there a config option to impose a timeout limit on SSL negotiation?

**Comment by brazilofmux** [comment]
It's a known issue now, and there is a related issue that needs attention as well. The problem is that while sockets are eventually set non-blocking, the SSL negotiation happens earlier than that, so it blocks. The solution is to go through, add more states to a connection, and be ready to repeat SSL calls like SSL_accept() or whatever as more data is available. Doable, but not done, yet.

**Comment by brazilofmux** [comment]
Non-blocking SSL is (after much pain and procrastination) in the queue.
